test/new-e2e/pkg/test/provisioner.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package test
+
+import "context"
+
+type RawResources map[string][]byte
+
+func (rr RawResources) Merge(in RawResources) {
+	for k, v := range in {
+		rr[k] = v
+	}
+}
+
+type Provisioner interface {
+	ID() string
+	Delete(string, context.Context) error
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Could be renamed to `TearDown` or `Cleanup`